pkg/openscap/openscap.go

```diff
@@ -0,0 +1,117 @@
+package openscap
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"syscall"
+)
+
+const (
+	CPE               = "oval:org.open-scap.cpe.rhel:def:"
+	CPE_DICT          = "/usr/share/openscap/cpe/openscap-cpe-oval.xml"
+	CVE_URL           = "https://www.redhat.com/security/data/metrics/ds/"
+	DIST_CVE_NAME_FMT = "com.redhat.rhsa-RHEL%d.ds.xml.bz2"
+	ARF_RESULT_FILE   = "results-arf.xml"
+	TMP_DIR           = "/tmp"
+)
+
+var (
+	RHEL_DIST_NUMBERS = [...]int{5, 6, 7}
+)
+
+func getRHELDist(chrootPath string) (int, error) {
+	for _, dist := range RHEL_DIST_NUMBERS {
+		output, err := oscapChroot(chrootPath, "oval", "eval", "--id",
+			fmt.Sprintf("%s%d", CPE, dist), CPE_DICT)
+		if err != nil {
+			return 0, err
+		}
+		if strings.Contains(string(output), fmt.Sprintf("%s%d: true", CPE, dist)) {
+			return dist, nil
+		}
+	}
+	return 0, fmt.Errorf("could not find RHEL dist")
+}
+
+func getInputCVE(cveDir string, dist int) (string, error) {
+	cveName := fmt.Sprintf(DIST_CVE_NAME_FMT, dist)
+	cveFileName := path.Join(cveDir, cveName)
+	cveURL, _ := url.Parse(CVE_URL)
+	cveURL.Path = path.Join(cveURL.Path, cveName)
+
+	out, err := os.Create(cveFileName)
+	if err != nil {
+		return "", fmt.Errorf("Could not create file %s: %v\n", cveFileName, err)
+	}
+	defer out.Close()
+
+	resp, err := http.Get(cveURL.String())
+	if err != nil {
+		return "", fmt.Errorf("Could not download file %s: %v\n", cveURL, err)
+	}
+	defer resp.Body.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	return cveFileName, err
+}
+
+func setOscapChrootEnv(chrootPath string) error {
+	os.Setenv("OSCAP_PROBE_ROOT", chrootPath)
+
+	out, err := exec.Command("uname", "-a").Output()
+	if err != nil {
+		return fmt.Errorf("Unable to run uname -a %v", err)
+	}
+	info := strings.Split(string(out), " ")
+	os.Setenv("OSCAP_PROBE_ARCHITECTURE", info[len(info)-1])
+	os.Setenv("OSCAP_PROBE_OS_NAME", info[0])
+	os.Setenv("OSCAP_PROBE_OS_VERSION", info[2])
```

Comment:
@enoodle @isimluk please come up with something that doesn't involve running `uname` to gather this.